src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -170,27 +170,27 @@ public function getRequestQuery()
 
     public function getRequestHeaders()
     {
-        return new HeaderBag($this->data['request_headers']);
+        return new ParameterBag($this->data['request_headers']);
```

Comment:
We are sure this breaks nothing right? In terms of possible unknown method calls by now (see also `getResponseHeaders`).